src/main/java/uk/gov/pay/publicauth/filters/LoggingFilter.java

```diff
@@ -27,22 +24,23 @@ public void init(FilterConfig filterConfig) { }
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) {
         Stopwatch stopwatch = Stopwatch.createStarted();
+        HttpServletRequest httpRequest = (HttpServletRequest)servletRequest;
 
-        String requestURL = ((HttpServletRequest) servletRequest).getRequestURI();
-        String requestMethod = ((HttpServletRequest) servletRequest).getMethod();
-        String requestId = StringUtils.defaultString(((HttpServletRequest) servletRequest).getHeader(HEADER_REQUEST_ID));
+        String requestURL = httpRequest.getRequestURI();
+        String requestMethod = httpRequest.getMethod();
+        String requestIdHeader = httpRequest.getHeader(HEADER_REQUEST_ID);
+        String requestId = requestIdHeader == null ? "" : requestIdHeader;
 
         MDC.put(HEADER_REQUEST_ID, requestId);
 
-        logger.info(format("[%s] - %s to %s began", requestId, requestMethod, requestURL));
+        logger.info("[{}] - {} to {} began", requestId, requestMethod, requestURL);
         try {
             filterChain.doFilter(servletRequest, servletResponse);
         } catch (Throwable throwable) {
-            logger.error("Exception - publicauth request - " + requestURL + " - exception - " + throwable.getMessage(), throwable);
+            logger.error("[{}] - Exception - {}", requestId, throwable.getMessage(), throwable);
         } finally {
-            logger.info(format("[%s] - %s to %s ended - total time %dms", requestId, requestMethod, requestURL,
-                    stopwatch.elapsed(TimeUnit.MILLISECONDS)));
-            stopwatch.stop();
+            logger.info("[{}] - {} to {} ended - total time {}ms", requestId, requestMethod, requestURL,
+                    stopwatch.stop().elapsed(TimeUnit.MILLISECONDS));
```

Comment:
Not sure about incorporating the `stopwatch.stop()` into logging statement because it means our logging statements have side effects

Reply:
Fair point. Undone.